jdaviz/app.py

```diff
@@ -1150,20 +1153,29 @@ def _get_multi_mask_subset_definition(self, subset_state):
                  "sky_region": None,
                  "subset_state": subset_state}]
 
-    def _get_roi_subset_definition(self, subset_state, to_sky=False):
-
-        # pixel region
-        roi_as_region = roi_subset_state_to_region(subset_state)
-
+    def _get_roi_subset_definition(self, subset_state, to_sky=False, wrt_data=None):
         wcs = None
-        if to_sky:
-            wcs = self._get_wcs_from_subset(subset_state)
+        if to_sky or wrt_data is not None:
+            wcs = self._get_wcs_from_subset(subset_state, data=wrt_data)
 
         # if no spatial wcs on subset, we have to skip computing sky region for this subset
         # but want to do so without raising an error (since many subsets could be requested)
         roi_as_sky_region = None
         if wcs is not None:
-            roi_as_sky_region = roi_as_region.to_sky(wcs)
+            if self.config == 'cubeviz' and wrt_data is None:
+                # If configuration is cubeviz and the SkyRegion must get accessed, use the first data
+                # loaded as a default option incase wrt_data is not set.
+                default_wcs = (self.data_collection[0].meta.get("_orig_spatial_wcs", None)
+                               if wrt_data is None else
+                               self.data_collection[wrt_data].meta.get("_orig_spatial_wcs", None))
+                roi_as_sky_region = roi_subset_state_to_region(subset_state).to_sky(default_wcs)
+            else:
+                roi_as_sky_region = roi_subset_state_to_region(subset_state, to_sky=True)
```

Comment:
This is needed in order to retrieve the `SkyRegion` as accessed by the subset's parent wcs (or the wcs from the first data loaded into cubeviz) to then convert it to a `PixelRegion` using the wcs of the data provided in `wrt_data`. Does that make sense @kecnry @cshanahan1 ?